src/Utils/multi_region_transformation.jl

```diff
@@ -114,6 +115,15 @@ Base.length(mo::MultiRegionObject)               = Base.length(mo.regional_objec
 Base.similar(mo::MultiRegionObject) = construct_regionally(similar, mo)
 Base.parent(mo::MultiRegionObject) = construct_regionally(parent, mo)
 
+on_architecture(::CPU, mo::MultiRegionObject) = MultiRegionObject(on_architecture(CPU(), mo.regional_objects))
+
+# TODO: Properly define on_architecture(::GPU, mo::MultiRegionObject) to handle cases where MultiRegionObject can be
+# distributed across different devices. Currently, the implementation assumes that all regional objects reside on a
+# single GPU.
+on_architecture(::GPU, mo::MultiRegionObject) = (
+MultiRegionObject(on_architecture(GPU(), mo.regional_objects);
+                  devices = Tuple(CUDA.device() for i in 1:length(mo.regional_objects))))
```

Comment:
I don't think we need to enclose the `MultiRegionObject` in parentheses; am I missing something?

```suggestion
on_architecture(::GPU, mo::MultiRegionObject) =
    MultiRegionObject(on_architecture(GPU(), mo.regional_objects);
                      devices = Tuple(CUDA.device() for i in 1:length(mo.regional_objects)))
```

Reply:
No, you aren't. Just committed your suggestion.